ldm/modules/kohya_lora_manager.py

```diff
@@ -0,0 +1,367 @@
+import re
+from pathlib import Path
+from ldm.invoke.devices import choose_torch_device
+from safetensors.torch import load_file
+import torch
+from torch.utils.hooks import RemovableHandle
+from diffusers.models import UNet2DConditionModel
+from transformers import CLIPTextModel
+
+'''
+This module supports loading LoRA weights trained with https://github.com/kohya-ss/sd-scripts
+To be removed once support for diffusers LoRA weights is well supported
+'''
+
+
+class LoRALayer:
+    lora_name: str
+    name: str
+    scale: float
+    up: torch.nn.Module
+    down: torch.nn.Module
+
+    def __init__(self, lora_name: str, name: str, rank=4, alpha=1.0):
+        self.lora_name = lora_name
+        self.name = name
+        self.scale = alpha / rank
```

Comment:
Alpha can be None or 0, that means unscalled layer.
Source:
https://github.com/kohya-ss/sd-scripts/blob/main/networks/lora.py#L21
https://github.com/kohya-ss/sd-scripts/blob/main/networks/lora.py#L51

Reply:
I'll take your word for it. Implemented.